stages/pod-fast.yaml

```diff
@@ -0,0 +1,124 @@
+kind: Stage
+apiVersion: kwok.x-k8s.io/v1alpha1
+metadata:
+  name: pod-create-and-ready
+spec:
+  resourceRef:
+    apiGroup: v1
+    kind: Pod
+  selector:
+    matchExpressions:
+      - key: '.metadata.deletionTimestamp'
+        operator: 'DoesNotExist'
+      - key: '.status.podIP'
+        operator: 'DoesNotExist'
+  next:
+    statusTemplate: |
+      {{ $now := Now }}
+      conditions:
+        - lastTransitionTime: {{ $now }}
+          status: "True"
+          type: Initialized
+        - lastTransitionTime: {{ $now }}
+          status: "True"
+          type: Ready
+        - lastTransitionTime: {{ $now }}
+          status: "True"
+          type: ContainersReady
+        {{ range .spec.readinessGates }}
+        - lastTransitionTime: {{ $now }}
+          status: "True"
+          type: {{ .conditionType }}
+        {{ end }}
+
+      containerStatuses:
+        {{ range .spec.containers }}
+        - image: {{ .image }}
+          name: {{ .name }}
+          ready: true
+          restartCount: 0
+          state:
+            running:
+              startedAt: {{ $now }}
+        {{ end }}
+
+      initContainerStatuses:
+        {{ range .spec.initContainers }}
+        - image: {{ .image }}
+          name: {{ .name }}
+          ready: true
+          restartCount: 0
+          state:
+            terminated:
+              exitCode: 0
+              finishedAt: {{ $now }}
+              reason: Completed
+              startedAt: {{ $now }}
+        {{ end }}
+
+      {{ with .status }}
+      hostIP: {{ with .hostIP }} {{ . }} {{ else }} {{ NodeIP }} {{ end }}
+      podIP: {{ with .podIP }} {{ . }} {{ else }} {{ PodIP }} {{ end }}
+      {{ end }}
+
+      phase: Running
+      startTime: {{ $now }}
+---
+kind: Stage
+apiVersion: kwok.x-k8s.io/v1alpha1
+metadata:
+  name: pod-completed-for-job
+spec:
+  resourceRef:
+    apiGroup: v1
+    kind: Pod
+  selector:
+    matchExpressions:
+      - key: '.metadata.deletionTimestamp'
+        operator: 'DoesNotExist'
+      - key: '.status.phase'
+        operator: 'In'
+        values:
+          - 'Running'
+      - key: '.metadata.ownerReferences.[].kind'
+        operator: 'In'
+        values:
+          - 'Job'
```

Comment:
If I want to simulate a plain Pod (not controlled by Job) to go through created->running->completed, I assume I can plumb particular labels to the Pod and match here, right?

Reply:
Yeah.